migrations/248_group_logs__indexes_add.sql

```diff
@@ -0,0 +1,6 @@
+-- Optimizes for "find all logs for a certain user in a certain group"
+-- queries that's used for things like generating the log of all assessment events.
+CREATE INDEX group_logs_group_id_user_id_idx ON group_logs (group_id, user_id);
+
+-- Optimizes for "find all logs for a certain user" queries.
+CREATE INDEX group_logs_user_id_idx ON group_logs (user_id);
```

Comment:
After rewriting the queries, I'm wondering if we actually just want to change to a single non-composite index on `group_id` - we no longer have any queries where we filter on `user_id` in a `JOIN` or `WHERE` clause.

If we don't change, I assume Postgres is smart enough to use the composite index even if we don't filter by `user_id`?

Reply:
Postgres is indeed smart enough to use the composite index for single-column lookups, but I'm also not opposed to just having a `(group_id)` index. I'm slightly inclined to keep the `(user_id)` index because I imagine that we'll want to look up logs by user at some point, but then again, YAGNI.

Reply:
👍 I'm going to change the multicolumn index to just be an index of `group_id`. I suspect that the cardinality of `group_logs` for a given group will always be small enough that we won't see much gain from including `user_id` on a multicolumn index.

I'll keep the `user_id` index around too.

Reply:
Done in ba3a9ac.

Reply:
👍